src/main/java/com/salesforce/dynamodbv2/mt/util/StreamArn.java

```diff
@@ -0,0 +1,211 @@
+package com.salesforce.dynamodbv2.mt.util;
+
+import static com.google.common.base.Preconditions.checkArgument;
+
+import com.salesforce.dynamodbv2.mt.mappers.MtAmazonDynamoDb.MtRecord;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * Stream ARN that includes a virtual table name in addition to the physical table name and stream label. See <a
+ * href="https://docs.aws.amazon.com/general/latest/gr/aws-arns-and-namespaces.html#arns-syntax">Amazon Resource Names
+ * (ARNs) and AWS Service Namespaces</a> for details on the ARN format.
+ */
+public class StreamArn {
+
+    static class MtStreamArn extends StreamArn {
+
+        private static final String VIRTUAL_FORMAT = "%s/context/%s/mttable/%s";
+
+        private final String context;
+        private final String mtTableName;
+
+        MtStreamArn(String partition, String service, String region, String accountId, String tableName,
+            String streamLabel, String context, String mtTableName) {
+            super(partition, service, region, accountId, tableName, streamLabel);
+            this.context = context;
+            this.mtTableName = mtTableName;
+        }
+
+        @Override
+        public Optional<String> getContext() {
+            return Optional.of(context);
+        }
+
+        @Override
+        public Optional<String> getMtTableName() {
+            return Optional.of(mtTableName);
+        }
+
+        @Override
+        public boolean matches(MtRecord record) {
+            return context.equals(record.getContext()) && mtTableName.equals(record.getTableName());
+        }
+
+        @Override
+        public boolean equals(Object o) {
+            if (this == o) {
+                return true;
+            }
+            if (o == null || getClass() != o.getClass()) {
+                return false;
+            }
+            if (!super.equals(o)) {
+                return false;
+            }
+            MtStreamArn that = (MtStreamArn) o;
+            return Objects.equals(context, that.context)
+                && Objects.equals(mtTableName, that.mtTableName);
+        }
+
+        @Override
+        public int hashCode() {
+            return Objects.hash(super.hashCode(), context, mtTableName);
+        }
+
+        @Override
+        public String toString() {
+            return String.format(VIRTUAL_FORMAT, super.toString(), context, mtTableName);
+        }
+    }
+
+    private static final String FORMAT = "arn:%s:%s:%s:%s:table/%s/stream/%s";
+
+    private static final Pattern ARN_PATTERN = Pattern.compile("arn:(?<partition>[^:]+):(?<service>[^:]+):"
```

Comment:
will this be (too) slow?

Reply:
I replaced it with manual parsing in latest commit.